### PR TITLE
Added ADSB to SpecPipe

### PR DESCRIPTION
**Pull request by omairsalam**
# Raw IQ Data Mocking and ADSB

## Streaming Mock Raw IQ Data

This example program streams the content of a file (`exampledata.bin`) to NATS Jetstream containg raw IQ Data collected at the frequency of 1090000000, sampling rate of 2000000 Hz, and gain of 50 collected for 10 seconds on a loop.

This circular streaming generates continuous Raw IQ Data without needing an actual antenna.

Side note: The Raw IQ data was collected using the following call (which requires rtl_sdr installed - not part of this code but added for more explanation):
```bash
rtl_sdr -f 1090000000 -s 2000000 -g 50 output.bin
```

All of this code is contained in the `mock_iq` folder

## Decoding Raw IQ to ADS-B

A client `main.py` has been created that fetches this Raw IQ Data from Jetstream, processes it via [Dump1090](https://github.com/antirez/dump1090), decodes the output and prints valid ADSB signals to the console.

### 1) Starting Mock Data Streamer (Terminal Window 1)
```
NATS_URL="nats://mytoken@127.0.0.1:4222"
DEVICE="dev1"
go run main.go exampledata.bin
```

We setup the NATS URL and DEVICE as environment variables that are used by `main.go`
We pass `exampledata.bin` to `main.go` so that's the data it plays on a loop

### 2) Processing Raw IQ to produce decoded ADSB Signals (Terminal Window 2)

```
pip install requirements.txt
NATS_URL="nats://127.0.0.1:4222"
DEVICE="dev1"
TOKEN="mytoken"
python3 main.py
```

### CODE REVIEW

File: _examples/adsb/dump1090
Comment (minghsu0107) [inline _examples/adsb/dump1090:1]: binary file should be removed

File: _examples/adsb/mock_iq/main.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+	f, err := os.Open(iqDataFilePath)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+	defer f.Close()
+
+	fmt.Printf("streaming raw IQ data to NATS subject %s\n", natsSubject)
+	chunk := make([]byte, 2*8192)
```
Comment (minghsu0107) [inline _examples/adsb/mock_iq/main.go:79]: chunk size is too small. SpecPipe uses 16*16384 bytes for IQ data

File: _examples/adsb/mock_iq/main.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+			case <-sig:
+				if err = publisher.Close(); err != nil {
+					fmt.Println(err)
+				}
+				return
+			default:
+			}
+		}
+
+		payload := make([]byte, 2*8192)
```
Comment (minghsu0107) [inline _examples/adsb/mock_iq/main.go:115]: same here for chunk size

File: _examples/adsb/mock_iq/main.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+			if err != nil {
+				if err == io.EOF {
+					f.Seek(0, 0)
+					continue
+				}
+				fmt.Println(err)
+				return
+			}
+			n += bytesRead
+			if n < 16384 {
```
Comment (minghsu0107) [inline _examples/adsb/mock_iq/main.go:102]: same here for chunk size

File: _examples/adsb/mock_iq/main.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+		if err != nil {
+			if err == io.EOF {
+				f.Seek(0, 0)
+				continue
+			}
+			fmt.Println(err)
+			return
+		}
+
+		for n < 16384 {
```
Comment (minghsu0107) [inline _examples/adsb/mock_iq/main.go:91]: same here for chunk size

File: _examples/adsb/mock_iq/mock_iq
Comment (minghsu0107) [inline _examples/adsb/mock_iq/mock_iq:1]: Binary file should not be included

File: _examples/adsb/main.py
Comment (minghsu0107) [inline _examples/adsb/main.py:1]: According to [PEP 8: Function and Variable Names](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names), function names should be [lowercase, with words separated by underscores](https://en.wikipedia.org/wiki/Snake_case) as necessary to improve readability. Variable names follow the same convention as function names.

File: _examples/adsb/utils.py
Comment (minghsu0107) [inline _examples/adsb/utils.py:1]: The util functions imported here do not follow the snake_case naming convention.

File: _examples/adsb/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+Side note: The Raw IQ data was collected using the following call (which requires rtl_sdr installed - not part of this code but added for more explanation):
+```bash
+rtl_sdr -f 1090000000 -s 2000000 -g 50 output.bin
+```
+
+All of this code is contained in the `mock_iq` folder
+
+## Decoding Raw IQ to ADS-B
+
+A client `main.py` has been created that fetches this Raw IQ Data from Jetstream, processes it via [Dump1090](https://github.com/antirez/dump1090), decodes the output and prints valid ADSB signals to the console.
```
[the remark this answers is not in the document]
Comment (omairsalam) [reply _examples/adsb/README.md:18]: I linked so the user can follow the original source of the information

**Pull request merged**